File: app/src/main/java/email/schaal/ocreader/api/json/v12/ItemMap.java
package email.schaal.ocreader.api.json.v12;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

import email.schaal.ocreader.database.model.Item;

/**
 * Aggregates feedIds and guidHashes, used to mark multiple items as starred
 */
public class ItemMap {
    private final Set<Map<String, Object>> items = new HashSet<>();

    public ItemMap(Iterable<Item> items) {
        for (Item item : items) {
            HashMap<String, Object> itemMap = new HashMap<>();
            itemMap.put("feedId", item.getFeedId());
            itemMap.put("guidHash", item.getGuidHash());
            this.items.add(itemMap);
        }
    }

    public Set<Map<String, Object>> getItems() {
        return items;
    }
}

File: app/src/main/java/email/schaal/ocreader/api/json/v12/ItemMap.kt
package email.schaal.ocreader.api.json.v12

import com.squareup.moshi.JsonClass
import email.schaal.ocreader.database.model.Item

/**
 * Aggregates feedIds and guidHashes, used to mark multiple items as starred
 */
@JsonClass(generateAdapter = true)
data class ItemMap(val items: List<MappedItem>) {
    @JsonClass(generateAdapter = true)
    data class MappedItem(val feedId: Long, val guidHash: String?)

    constructor(sourceItems: Iterable<Item>) : this(sourceItems.map {
        MappedItem(it.feedId, it.guidHash)
    })
}